components/collections/codepointtrie_builder/cpp/README.md

```diff
@@ -1,108 +1,23 @@
 # CodePointTrie Builder Tool
 
-This directory contains C++ bindings to the ICU4C CodePointTrie builder in the form of an executable named `list_to_ucptrie`. There are two ways to run this tool:
-
-1. As a WebAssembly module
-2. As a native executable
+This directory contains C++ bindings to the ICU4C CodePointTrie builder in the form of a wrapper named `ucptrie_wrap`. This tool is intended to be used as a shared library in WebAssembly.
 
 The WebAssembly module is checked into tree and made available to Rust library clients.
 
-## Usage
-
-`list_to_ucptrie` takes line-separated integers in STDIN, corresponding to the trie values for code points 0, 1, 2, and so on. When it receives an EOF (`^D`), it prints helpful information to STDERR and dumps the CodePointTrie TOML to STDOUT.
-
-It takes three positional arguments:
-
-1. Default value for code points (integer)
-2. Error value for out-of-range code points (integer)
-3. TrieType: `fast` or `small`
-
 ## WebAssembly Module
 
 To build the WebAssembly module, you need:
 
 - Local copy of the ICU4C sources (May 6, 2022 or later: [eea7985](https://github.com/unicode-org/icu/commit/eea7985e5a7108d00f1224ed36f0220fa9441cdc))
-- [wasienv](https://github.com/wasienv/wasienv)
+- The following packages which can be fetched from apt-get:
+  - `lld`
+  - `libc++-dev-wasm32`
+  - `libclang-rt-dev-wasm32`
 
 Once you have these two tools installed, from this directory, simply run:
 
 ```bash
-$ make ICU4C_SOURCE=/path/to/icu4c/source list_to_ucptrie.wasm
-```
-
-To test the WASM file, you can use `wasirun`:
-
-```bash
-$ wasirun list_to_ucptrie.wasm
-Takes 3 positional arguments: default value, error value, and trie type
+$ make ICU4C_SOURCE=/path/to/icu4c/source ucptrie_wrap.wasm
 ```
 
 You can then copy the wasm file up one directory in order to update the version shipped with ICU4X.
-
-## Native Executable
```

Comment:
issue: why are you deleting the native instructions?

Reply:
They are obsolete. Previously I was expecting that we would use the same executable as either a WASM+WASI executable or a native executable with the only difference being how we plug the stdin/stdout. However, the native bindings for this crate ended up treating ICU4C as a system library without needing to build a separate C++ executable. I have therefore changed this code from being an executable named `list_to_ucptrie` to being a library named `ucptrie_wrap`.